### PR TITLE
ci: fixed detox tests.

### PR DESCRIPTION
**Pull request by zhyd1997**
Close: https://github.com/jhipster/generator-jhipster-react-native/issues/889

### CODE REVIEW

**Comment by zhyd1997** [comment]
Will update again in the weekend.

**Comment by zhyd1997** [comment]
Still something wrong on e2e:

Screenshot 2023-12-02 at 17 38 51

File: test/scripts/generate-react-native-app.sh [hunk 1]
```diff
@@ -16,5 +16,8 @@ else
   --defaults --no-insight --skip-git
 fi
 
+# pin specific version of expo
+npm i expo@48.0.6
+
```
Comment (zhyd1997) [inline test/scripts/generate-react-native-app.sh:21]: There's an issue in the latest `expo`, which mean `e2e` will fails in the local env as well:

Screenshot 2023-12-03 at 10 44 05

https://github.com/expo/expo/issues/25551

Comment (zhyd1997) [reply test/scripts/generate-react-native-app.sh:21]: But it failed:

Screenshot 2023-12-03 at 19 47 48

**Comment by zhyd1997** [comment]
<details>
  <summary><b>Update</b>: Failed on the real machine (Mac OS 14.1.2 (23B92)):</summary>
  Screenshot 2023-12-05 at 23 33 50
</details>

**Comment by zhyd1997** [comment]
Just some interesting findings:

Circle CI works for now (Xcode `v15.1.0` (latest) - green):

Screenshot 2023-12-17 at 20 26 25

but e2e CI failed in github actions (Xcode `v15.0.1` (latest) - timeout)

@mraible @mshima **PTAL**

**Comment by zhyd1997** [comment]
I will download the artifacts to see what’s happened in the tests, and will update the tests in the weekend if possible.

**Comment by zhyd1997** [comment]
Hi @mraible @mshima 

Could you please take a look if the pull request makes sense? :eyes:

Although there's something break in the **e2e** test files, eg, some issues I have found:

1. Album access permission (or the tests will hang out due to the prompt)

<details>
  <summary>Expand to see the screenshot</summary>
  Simulator Screenshot - iPhone 15 - 2023-12-19 at 20 37 05
</details>


2. Strange Hard-coded element
Screenshot 2023-12-19 at 20 35 25


IMO the CI works now (see OAuth App).

**Comment by mshima** [comment]
It's already an improvement.
But but tests should pass so we can add as requirement to pass otherwise dependabot can break it again.
For now I recommend to skip failing tests.

**Comment by mshima** [comment]
You should use `describe.skip` instead of filter test.

**Comment by zhyd1997** [comment]
Hi @mshima @mraible 

I have upgraded Xcode from 14 to 15.1 (latest version currently) because:

1. all of tests are passed on my machine using XCode 15.1.0 and IOS simulator iPhone 15+.
2. we always maintain dependencies but not for e2e test enviroment, so I handled it in this pull request.

Please let me know if you have any questions!

Merry Chrismas! 🎄🧑‍🎄

**Comment by zhyd1997** [comment]
and seems that Circle CI is more reliable than Github Actions.

Screenshot 2023-12-23 at 20 42 39

**Comment by mraible** [comment]
@zhyd1997 Can you please fix the conflicts in this PR?

**Comment by mshima** [comment]
I thought I had merged this PR =\

**Pull request merged**